src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -71,16 +80,20 @@ private void processBlob(BlobClient blob) {
         envelopeService
             .findEnvelope(blob.getBlobName(), blob.getContainerName())
             .ifPresentOrElse(
-                envelope -> logger.info(
-                    "Envelope already processed in system, skipping."
-                        + " ID: {}, filename: {}, container: {}, state: {}",
-                    envelope.id,
-                    envelope.fileName,
-                    envelope.container,
-                    envelope.status.name()
-                ),
+                envelope -> {
+                    if (isStale(envelope)) {
+                        logger.warn("Found stale envelope. Resuming its processing. {}", envelope.getBasicInfo());
+                        blobProcessor.continueProcessing(envelope.id, blob);
+                    } else {
+                        logger.info("Envelope already processed in system, skipping. {}", envelope.getBasicInfo());
+                    }
+                },
                 () -> blobProcessor.process(blob)
             );
+    }
 
+    private boolean isStale(Envelope envelope) {
```

Comment:
Do we need this notion of stale envelope in the context of processing? I though a lease on the blob is what should give you access to the envelope in `CREATED` status.

Reply:
good point if you acquire lease we should process it then.

Reply:
fixed